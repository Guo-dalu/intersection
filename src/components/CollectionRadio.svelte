<script lang="ts">
	import { Radio, P } from 'flowbite-svelte'
	import Punchline from './Punchline.svelte'
	export let iterateCollection = 'A'
</script>

{#each ['A', 'B'] as collection}
	<div class="w-max mx-auto mt-3">
		<Radio name="collections" value={collection} bind:group={iterateCollection}>
			<P class="font-light">Collection {collection}</P>
		</Radio>
	</div>
{/each}
<P class="mt-4 font-light text-center"
	>Then the Collection <Punchline>{iterateCollection === 'A' ? 'B' : 'A'}</Punchline> would be put into
	the Hash Set
</P>
