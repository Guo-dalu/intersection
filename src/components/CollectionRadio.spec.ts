import { expect, describe, it } from 'vitest'
import { render } from '@testing-library/svelte'
import CollectionRadio from './CollectionRadio.svelte'

describe('CollectionRadio Component', () => {
	it('should render correctly', () => {
		const { container } = render(CollectionRadio, { iterateCollection: 'B' })

		expect(container).toMatchSnapshot()
	})
})
