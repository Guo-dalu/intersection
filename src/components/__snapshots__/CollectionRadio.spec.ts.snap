// Vitest Snapshot v1

exports[`CollectionRadio Component > should render correctly 1`] = `
<body>
  <div>
    <div
      class="w-max mx-auto mt-3"
    >
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300 flex items-center"
      >
        <input
          class="w-4 h-4 bg-gray-100 border-gray-300 dark:ring-offset-gray-800 focus:ring-2 mr-2 dark:bg-gray-700 dark:border-gray-600 text-primary-600 focus:ring-primary-500 dark:focus:ring-primary-600"
          name="collections"
          type="radio"
          value="A"
        />
        <p
          class="text-base text-gray-900 dark:text-white leading-normal text-left whitespace-normal font-light"
        >
          Collection 
          A
        </p>
        <!--&lt;P&gt;-->
      </label>
      
      <!--&lt;Label&gt;-->
      <!--&lt;Radio&gt;-->
    </div>
    <div
      class="w-max mx-auto mt-3"
    >
      <label
        class="text-sm font-medium text-gray-900 dark:text-gray-300 flex items-center"
      >
        <input
          class="w-4 h-4 bg-gray-100 border-gray-300 dark:ring-offset-gray-800 focus:ring-2 mr-2 dark:bg-gray-700 dark:border-gray-600 text-primary-600 focus:ring-primary-500 dark:focus:ring-primary-600"
          name="collections"
          type="radio"
          value="B"
        />
        <p
          class="text-base text-gray-900 dark:text-white leading-normal text-left whitespace-normal font-light"
        >
          Collection 
          B
        </p>
        <!--&lt;P&gt;-->
      </label>
      
      <!--&lt;Label&gt;-->
      <!--&lt;Radio&gt;-->
    </div>
     
    <p
      class="text-base text-gray-900 dark:text-white leading-normal whitespace-normal mt-4 font-light text-center"
    >
      Then the Collection 
      <span
        class="gradientcolor s-qaK2xGRD7wDi"
      >
        A
      </span>
      <!--&lt;Punchline&gt;-->
       would be put into
	the Hash Set
    </p>
    <!--&lt;P&gt;-->
    <!--&lt;CollectionRadio&gt;-->
  </div>
</body>
`;
